src/Microphysics2M.jl

```diff
@@ -219,15 +293,170 @@ function rain_self_collection(
         return FT(0)
     end
 
-    kr::FT = CMP.kr_SB2001(param_set)
+    krr::FT = CMP.krr_SB2006(param_set)
+    κrr::FT = CMP.κrr_SB2006(param_set)
+    d::FT = CMP.d_sc_SB2006(param_set)
+    ρw::FT = CMP.ρ_cloud_liq(param_set)
+    ρ0::FT = CMP.ρ0_SB2006(param_set)
 
     L_rai = ρ * q_rai
+    λr =
+        lambda_rai_limited(param_set, L_rai, N_rai) *
+        (SF.gamma(4) / FT(π) / ρw)^FT(1 / 3)
 
-    dN_rai_dt_sc = -kr * N_rai * L_rai
+    dN_rai_dt_sc = -krr * N_rai * L_rai * sqrt(ρ0 / ρ) * (1 + κrr / λr)^d
 
     return dN_rai_dt_sc
 end
 
+"""
+    rain_breakup(param_set, scheme, q_rai, ρ, dN_rai_dt_sc)
+
+ - `param_set` - abstract set with Earth parameters
+ - `scheme` - type for 2-moment liquid self-collection parameterization
+ - `q_rai` - rain water specific humidity
+ - `ρ` - air density
+ - `N_rai` - raindrops number density
+ - `dN_rai_dt_sc` - rate of change of raindrops number density due to self-collection
+
+Returns the raindrops number density tendency due to breakup of raindrops
+that produce smaller raindrops for `scheme == SB2006Type`
+"""
+function rain_breakup(
+    param_set::APS,
+    scheme::CT.SB2006Type,
+    q_rai::FT,
+    ρ::FT,
+    N_rai::FT,
+    dN_rai_dt_sc::FT,
+) where {FT <: Real}
+
+    if q_rai < eps(FT)
+        return FT(0)
+    end
+
+    ρw::FT = CMP.ρ_cloud_liq(param_set)
+    Deq::FT = CMP.Deq_br_SB2006(param_set)
+    Dr_th::FT = CMP.Dr_th_br_SB2006(param_set)
+    kbr::FT = CMP.kbr_SB2006(param_set)
+    κbr::FT = CMP.κbr_SB2006(param_set)
+
+    L_rai = ρ * q_rai
+    xr = mean_mass_rai_limited(param_set, L_rai, N_rai)
+    Dr = (xr * 6 / π / ρw)^FT(1 / 3)
+    ΔD = Dr - Deq
+    phi_br =
+        (Dr < Dr_th) ? FT(-1) : ((ΔD <= 0) ? kbr * ΔD : 2 * (exp(κbr * ΔD) - 1))
```

Comment:
I saw your note in the docs, but to me it looks like it's still not contuntous?

In the linear regime I have `1000 * 0.9 * 1e-3` vs in the exponential regime `2 * (exp(2.3 * 1e3 * 0.9 * 1e-3) -1) = 13.849646235698973`?

Reply:
> I saw your note in the docs, but to me it looks like it's still not contuntous?
> 
> In the linear regime I have `1000 * 0.9 * 1e-3` vs in the exponential regime `2 * (exp(2.3 * 1e3 * 0.9 * 1e-3) -1) = 13.849646235698973`?

In both linear and exponential functions, the input is Delta D which is D - Deq. So in the linear regime `1000 * (0.9 -0.9) =0` equal `2*(exp(2.3 * 1e3 * (0.9 - 0.9)) - 1) = 0` in the exponential regime.